pvnet/models/multimodal/validation/config_static_validation.py

```
@@ -0,0 +1,514 @@
+        context: Context for error messages.
+        source_key: Sub-key if section_key points to a dict of encoders (e.g., NWP source).
+
+    Raises:
+        KeyError, TypeError, ValueError: If parameters are missing, wrong type, or not positive.
+    """
+    convnet_params = [
+        "in_channels",
+        "out_features",
+        "number_of_conv3d_layers",
+        "conv3d_channels",
+        "image_size_pixels",
+    ]
+    encoder_config = get_encoder_config(cfg, section_key, context, source_key)
+    _check_positive_int_params_in_dict(encoder_config, convnet_params, context)
+
+
+def _check_attention_encoder_params(
+    cfg: dict[str, Any], section_key: str, context: str
+) -> None:
+    """Validate required positive integer parameters for Attention-based encoders.
+
+    Args:
+        cfg: The main configuration dictionary.
+        section_key: Key of the encoder section (e.g., 'pv_encoder').
+        context: Context for error messages.
+
+    Raises:
+        KeyError, TypeError, ValueError: If parameters are missing, wrong type, or not positive.
+    """
+    attention_params = [
+        "num_sites",
+        "out_features",
+        "num_heads",
+        "kdim",
+        "id_embed_dim",
+    ]
+    encoder_config = get_encoder_config(cfg, section_key, context, None)
+    _check_positive_int_params_in_dict(encoder_config, attention_params, context)
+
+
+def _check_dict_values_are_int(
+    data: dict[str, Any],
+    dict_name: str,
+) -> None:
+    """Check if all values in a dictionary are integers (logs warning on mismatch)."""
+    expected_type = int
+    for source, value in data.items():
+        if not isinstance(value, expected_type):
+            message = (
+                f"'{dict_name}' for source '{source}' expected {expected_type.__name__}, "
+                f"found {type(value).__name__}."
+            )
+            logger.warning(message)
```

I think this is only used for nwp parameters and should fail hard because we require all of them to be ints